### PR TITLE
Debugger: Various fixes

### PR DESCRIPTION
**Pull request by SuperSamus**
Fixes many annoying bugs and inconsistencies when using the debugger:
- Run to Here now actually does what it says on the name (fixes https://bugs.dolphin-emu.org/issues/13532). It also can't be used without a game running.
- You can create breakpoints while a game isn't running (the button isn't disabled anymore).
- Some cleanups to the API:
  - Misleading names were changed
  - Functions were used where it makes sense, less pointless reimplementations
    - Notably, Step Out and the Interpreter don't manually check for breakpoints, but use the same implementation as the Cached Interpreter and JIT. This means that they don't break on log-only breakpoints anymore.
- Pressing Continue while on top of a breakpoint did nothing. Now it... does something! (It resumes, like on Cached Interpreter and JIT)

Also, temporary breakpoints have been changed. Before:
1. In theory there could be multiple, but in practice they were (manually) cleared before creating one
2. (Some of) the conditions to clear one were either to reach it, to create a new one (due to the point above), or to step. This created weird behavior: let's say you Step Over a `bl` (thus creating a temporary breakpoint on `pc+4`), and you reached a regular breakpoint inside the `bl`. The temporary one would still be there: if you resumed, the emulation would still stop there, as a sort of Step Out. But, if before resuming, you made a Step, then it wouldn't do that.
3. The breakpoint widget had no idea concept of them, and will treat them as regular breakpoints. Also, they'll be shown only when the widget is updated in some other way, leading to more confusion.
4. Because only one breakpoint could exist per address, the creation of a temporary breakpoint on a top of a regular one would delete it and inherit its properties (e.g. being log-only). This could happen, for instance, if you Stepped Over a `bl` specifically, and pc+4 had a regular breakpoint.

Now there can only be one temporary breakpoint, which is automatically cleared whenever emulation is paused. So, removing some manual clearing from 1., and removing the weird behavior of 2. As it is stored in a separate variable, it won't be seen at all depending on the function used (fixing 3., and removing some checks in other places), and it won't replace a regular breakpoint, instead simply having priority (fixing 4.).

Now, things are still not perfect on the API level:
- There's nothing "forcing" whoever sets a temporary breakpoint to restart execution immediately
- There's obviously more that could be done, like more code reuse, and moving some functions (there are many implementations on the widgets themselves).

Still, it should be better than before.

### CODE REVIEW

**Comment by InusualZ** [comment]
I'm talking from experience, you need to split the changes into different commit before you would get a review.

**Comment by JosJuice** [comment]
This would be much easier to review if it was split up, yes.

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/Core.cpp [hunk 1]
```diff
@@ -709,13 +709,13 @@ void SetState(Core::System& system, State state, bool report_state_change)
...
     //   stopped (including the CPU).
-    system.GetCPU().EnableStepping(true);  // Break
+    system.GetCPU().SetStepping(true);  // Break
     Wiimote::Pause();
     ResetRumble();
     break;
   case State::Running:
   {
-    system.GetCPU().EnableStepping(false);
+    system.GetCPU().SetStepping(false);
```
Comment (JosJuice) [inline Source/Core/Core/Core.cpp:718]: Please also update the commend in this file that mentions EnableStepping.

**Comment by SuperSamus** [comment]
I messed up which commit touches which comment...

**Comment by TryTwo** [comment]
It's a definitely improvement.  Nothing odd jumped out at me.  I don't know the inner working of Interpreter, but are you sure "step before checking" won't produce unexpected behavior at times?

**Comment by SuperSamus** [comment]
I didn't notice anything weird.
Still, I changed the way the last commit does its thing. It's probably a more elegant solution.
(I also added a separate `CheckBreakPoints` in the second commit.)

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

**Comment by SuperSamus** [comment]
Rebased to fix conflicts.

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

**Comment by dreamsyntax** [comment]
@JMC47 from my own usage this seems good to me.

**Review by AdmiralCurtiss — APPROVED** [review]
Looks good.

**Pull request merged**